numpyro/distributions/continuous.py

```diff
@@ -1033,6 +1033,115 @@ def icdf(self, q):
         return self.loc + self.scale * logit(q)
 
 
+def _kron(A, B):
+    D = A[..., :, None, :, None] * B[..., None, :, None, :]
+    ds = D.shape
+    newshape = (*ds[:-4], ds[-4] * ds[-3], ds[-2] * ds[-1])
+    return D.reshape(newshape)
+
+
+class MatrixNormal(Distribution):
+    arg_constraints = {
+        "loc": constraints.real_vector,
+        "scale_columns": constraints.positive_definite,
+        "scale_rows": constraints.positive_definite,
+    }
+    support = constraints.real_matrix
+    reparametrized_params = ["loc", "scale_columns", "scale_rows"]
+
+    def __init__(self, loc, scale_columns, scale_rows, validate_args=None):
+        """_summary_
+
+        Args:
+            loc (_type_): _description_
+            scale_columns (_type_): _description_
+            scale_rows (_type_): _description_
+            validate_args (_type_, optional): _description_. Defaults to None.
+
+        Raises:
+            ValueError: _description_
+        """
+        if not (loc.ndim == scale_columns.ndim == scale_rows.ndim):
+            raise ValueError(
+                "ndim of loc, scale_columns and scale_rows are reuqired to be equal."
+            )
+
+        event_shape = loc.shape[-2:]
+        n, p = event_shape
+        batch_shape = loc.shape[:-2]
+
+        assert scale_rows.shape == batch_shape + (
+            n,
+            n,
+        ), "scale_rows.shape does not match"
+        assert scale_columns.shape == batch_shape + (
+            p,
+            p,
+        ), "scale_rows.shape does not match"
+
+        super(MatrixNormal, self).__init__(
+            batch_shape=batch_shape, event_shape=event_shape
+        )
+
+        self.loc = loc
+        self.scale_columns = scale_columns
+        self.scale_rows = scale_rows
+
+    @property
+    def mean(self):
+        return self.loc
+
+    def sample(self, key, sample_shape=()):
+
+        mvn_dim = np.prod(self.event_shape)
+        loc = jnp.reshape(self.loc, newshape=self.batch_shape + (mvn_dim,), order="F")
+        assert loc.shape == self.batch_shape + (mvn_dim,), "vec shape does not match"
+
+        cov = _kron(self.scale_rows, self.scale_columns)
+        assert cov.shape == self.batch_shape + (
+            mvn_dim,
+            mvn_dim,
+        ), "cov shape does not match"
+
+        mvn = MultivariateNormal(loc=loc, covariance_matrix=cov)
+
+        assert mvn.batch_shape == self.batch_shape, "batch shape of MVN does not match"
+        assert mvn.event_shape == (
+            np.prod(self.event_shape),
+        ), "event shape of MVN does not match"
+
+        samples_mvn = mvn.sample(key, sample_shape=sample_shape)
```

Comment:
This seems expensive. Could we simply use the procedure in [this section](https://en.wikipedia.org/wiki/Matrix_normal_distribution#Drawing_values_from_the_distribution)?
```
eps = random.normal(
       key, shape=sample_shape + self.batch_shape + self.event_shape)
return self.loc + self.scale_tril_row @ eps @ jnp.swapaxes(self.scale_tril_column, -2, -1)
```

Reply:
Indeed. This is much simpler.